src/Frame.cpp

```diff
@@ -818,16 +818,20 @@ void Frame::AddImage(std::shared_ptr<QImage> new_image, bool only_odd_lines)
 		// Ignore image of different sizes or formats
 		bool ret=false;
 		#pragma omp critical (AddImage)
-		if (image == new_image || image->size() != image->size() || image->format() != image->format())
-			ret=true;
-		if (ret)
-			return;
```

Comment:
Ignoring the weird use of the `ret` boolean here, what's up with those _conditions_!?!?

`if ( image->size() != image->size() )`!? `if ( image->format() != image->format() )`!?

Clearly the first was supposed to be `image->size() != new_image->size()`, so I fixed it to that. The second was presumably supposed to be `image->format() != new_image->format()`, but since we can format-convert with QImage, instead of doing that test I convert `new_image` to the format of `image` if it doesn't match.

Reply:
> Ignoring the weird use of the `ret` boolean here,

(Aha, it seems there's a reason for _that_, due to OpenMP's weird semantics.)